R/plot-build.r

```diff
@@ -22,7 +22,13 @@ ggplot_build <- function(plot) {
   }
 
   layers <- plot$layers
-  layer_data <- lapply(layers, function(y) y$data)
```

Comment:
This isn't the right place either - if you wanted to preserve this basic approach it would be better to add a `get_data()` method (probably needs a better name) to the layer object.